interfaces/builtin/personal_files.go

```diff
@@ -76,3 +80,63 @@ func init() {
 		},
 	})
 }
+
+// potentiallyMissingDirs returns an ensure directory specification that contains the information
+// required to create potentially missing directories for the given raw path. Potentially missing
+// directories are those that are implicitly required in order to enable the user to create
+// the target directory specified in a write attribute path.
+func potentiallyMissingDirs(rawPath interface{}) (*interfaces.EnsureDirSpec, error) {
+	path, ok := rawPath.(string)
+	if !ok {
+		// BeforePreparePlug should to prevent this
+		return nil, fmt.Errorf("%[1]v (%[1]T) is not a string", rawPath)
+	}
+
+	// All directories between $HOME and the leaf directory are potentially missing
+	path = filepath.Clean(path)
+	pathElements := strings.Split(path, string(filepath.Separator))
+	if pathElements[0] != "$HOME" {
+		// BeforePreparePlug should prevent this
+		return nil, fmt.Errorf(`%q must start with "$HOME/"`, path)
+	}
+	if len(pathElements) < 3 {
+		return nil, nil
+	}
+
+	ensureDirSpec := &interfaces.EnsureDirSpec{
+		// EnsureDir prefix directory that must exist
+		MustExistDir: pathElements[0],
+		// Directory to ensure by creating the missing directories within MustExistDir
+		EnsureDir: filepath.Join(pathElements[:len(pathElements)-1]...),
+	}
+	return ensureDirSpec, nil
```

Comment:
nitpick: slightly more concise
```suggestion
	return &interfaces.EnsureDirSpec{
		// EnsureDir prefix directory that must exist
		MustExistDir: pathElements[0],
		// Directory to ensure by creating the missing directories within MustExistDir
		EnsureDir: filepath.Join(pathElements[:len(pathElements)-1]...),
	}, nil
```

Reply:
Thanks, improved